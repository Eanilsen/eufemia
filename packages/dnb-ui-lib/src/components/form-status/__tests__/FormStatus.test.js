/**
 * Component Test
 *
 */

import React from 'react'
import {
  mount,
  fakeProps,
  axeComponent,
  toJson,
  loadScss
} from '../../../core/jest/jestSetup'
import Component from '../FormStatus'

// just to make sure we re-run the test in watch mode due to changes in theese files
import _form_status from '../style/_form-status.scss' // eslint-disable-line
import dnb_form_status from '../style/dnb-form-status.scss' // eslint-disable-line
import dnb_form_status_theme_ui from '../style/themes/dnb-form-status-theme-ui.scss' // eslint-disable-line

const props = fakeProps(require.resolve('../FormStatus'), {
  optional: true
})
props.state = 'error'
props.status = null
props.hidden = false
props.icon = 'exclamation'

describe('FormStatus component', () => {
  const Comp = mount(<Component {...props} />)

  it('have to match snapshot', () => {
    expect(toJson(Comp)).toMatchSnapshot()
  })

  it('should validate with ARIA rules', async () => {
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })

  it('should have correact attributes once the "hidden" prop changes', async () => {
    const Comp = mount(<Component {...props} hidden />)
    expect(Comp.exists('[aria-hidden]')).toBe(true)
    expect(Comp.exists('[aria-live="assertive"]')).toBe(false)
    Comp.setProps({
      hidden: false
    })
    expect(Comp.exists('[aria-live="assertive"]')).toBe(true)
    expect(await axeComponent(Comp)).toHaveNoViolations()
  })

  it('has to to have a text value as defined in the prop', () => {
    expect(Comp.find('.dnb-form-status--text').text()).toBe('text')
  })
})

describe('FormStatus scss', () => {
  it('have to match snapshot', () => {
    const scss = loadScss(require.resolve('../style/dnb-form-status.scss'))
    expect(scss).toMatchSnapshot()
  })
  it('have to match default theme snapshot', () => {
    const scss = loadScss(
      require.resolve('../style/themes/dnb-form-status-theme-ui.scss')
    )
    expect(scss).toMatchSnapshot()
  })
})
