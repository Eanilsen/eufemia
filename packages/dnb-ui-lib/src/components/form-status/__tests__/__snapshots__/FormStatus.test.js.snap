// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormStatus component have to match snapshot 1`] = `
<FormStatus
  animation="animation"
  class="class"
  className="className"
  hidden={false}
  icon="exclamation"
  icon_size="icon_size"
  render_content={[Function]}
  status="error"
  text="text"
  text_id="text_id"
  title="title"
>
  <span
    aria-live="assertive"
    className="dnb-form-status dnb-form-status--error dnb-form-status--animation dnb-form-status--has-content className class"
    hidden={false}
    title="title"
  >
    <IconPrimary
      alt={null}
      aria-hidden={true}
      attributes={null}
      class={null}
      className={null}
      color={null}
      height={null}
      icon="exclamation"
      modifier={null}
      size="icon_size"
      title={null}
      width={null}
    >
      <span
        alt="exclamation"
        aria-hidden={true}
        className="dnb-icon"
        role="img"
      >
        <exclamation>
          <svg
            fill="none"
            height={16}
            viewBox="0 0 16 16"
            width={16}
          >
            <path
              clipRule="evenodd"
              d="M8 1.75a.75.75 0 0 1 .75.75v7a.75.75 0 0 1-1.5 0v-7A.75.75 0 0 1 8 1.75zm-.893 10.357A1.25 1.25 0 0 1 8 11.75c.32 0 .646.11.893.357s.357.573.357.893c0 .32-.11.646-.357.893A1.249 1.249 0 0 1 8 14.25c-.32 0-.646-.11-.893-.357A1.249 1.249 0 0 1 6.75 13c0-.32.11-.646.357-.893z"
              fill="#000"
              fillRule="evenodd"
            />
          </svg>
        </exclamation>
      </span>
    </IconPrimary>
    <span
      className="dnb-form-status--text"
      id="text_id"
    >
      text
    </span>
  </span>
</FormStatus>
`;

exports[`FormStatus scss have to match default theme snapshot 1`] = `
"/*
 * FormStatus theme
 *
 */
.dnb-form-status {
  font-size: var(--font-size-small); }
  .dnb-form-status--error {
    color: var(--color-cherry-red);
    background-color: var(--color-cherry-red-80); }
  .dnb-form-status--info {
    color: var(--color-emerald-green);
    background-color: var(--color-mint-green-25); }
"
`;

exports[`FormStatus scss have to match snapshot 1`] = `
"/*
* DNB FormStatus
*
*/
/**
 * This file is only used to make components independent
 * so that they can get imported individually, without the core styles
 *
 */
/*
 * Utilities
 */
/*
 * Scopes
 *
 */
/*
 * Document Reset
 *
 */
.dnb-form-status {
  /**
   * 1. Remove repeating backgrounds in all browsers (opinionated).
   * 2. Add border box sizing in all browsers (opinionated).
   */
  /**
   * 1. Add text decoration inheritance in all browsers (opinionated).
   * 2. Add vertical alignment inheritance in all browsers (opinionated).
   */
  /* Grouping content
   */
  /**
   * 1. Add the correct sizing in Firefox.
   * 2. Show the overflow in Edge and IE.
   */
  /**
   * Add the correct display in IE.
   */
  /**
   * Remove the list style on navigation lists in all browsers (opinionated).
   */
  /**
   * 1. Use the default monospace user interface font
   *    in all browsers (opinionated).
   * 2. Correct the odd \`em\` font sizing in all browsers.
   */
  /* Text-level semantics
   */
  /**
   * Remove the gray background on active links in IE 10.
   */
  /**
   * Add the correct text decoration in Edge, IE, Opera, and Safari.
   */
  /**
   * Add the correct font weight in Chrome, Edge, and Safari.
   */
  /**
   * 1. Use the default monospace user interface font
   *    in all browsers (opinionated).
   * 2. Correct the odd \`em\` font sizing in all browsers.
   */
  /*
   * Remove the text shadow on text selections in Firefox 61- (opinionated).
   * 1. Restore the coloring undone by defining the text shadow
   *    in all browsers (opinionated).
   */
  /* Embedded content
   */
  /*
   * Change the alignment on media elements in all browers (opinionated).
   */
  /**
   * Add the correct display in IE 9-.
   */
  /**
   * Add the correct display in iOS 4-7.
   */
  /**
   * Remove the border on images inside links in IE 10-.
   */
  /**
   * Change the fill color to match the text color in all browsers (opinionated).
   */
  /**
   * Hide the overflow in IE.
   */
  /* Tabular data
   */
  /**
   * Collapse border spacing in all browsers (opinionated).
   */
  /* Forms
   */
  /**
   * Inherit styling in all browsers (opinionated).
   */
  /**
   * Remove the margin in Safari.
   */
  /**
   * 1. Show the overflow in IE.
   * 2. Remove the inheritance of text transform in Edge, Firefox, and IE.
   */
  /**
   * Correct the inability to style clickable types in iOS and Safari.
   */
  /**
   * Correct the padding in Firefox.
   */
  /**
   * Show the overflow in Edge and IE.
   * 1. Safari did inherit the break-word from html root
   */
  /**
   * 1. Correct the text wrapping in Edge and IE.
   * 2. Correct the color inheritance from \`fieldset\` elements in IE.
   */
  /**
   * 1. Add the correct display in Edge and IE.
   * 2. Add the correct vertical alignment in Chrome, Firefox, and Opera.
   */
  /**
   * Remove the inheritance of text transform in Firefox.
   */
  /**
   * 2. Remove the default vertical scrollbar in IE.
   * 3. Change the resize direction on textareas in all browsers (opinionated).
   */
  /**
   * Remove the padding in IE 10-.
   */
  /**
   * 1. Correct the odd appearance in Chrome and Safari.
   * 2. Correct the outline style in Safari.
   */
  /**
   * Correct the cursor style of increment and decrement buttons in Safari.
   */
  /**
   * Correct the text style of placeholders in Chrome, Edge, and Safari.
   */
  /**
   * Remove the inner padding in Chrome and Safari on macOS.
   */
  /**
   * 1. Correct the inability to style clickable types in iOS and Safari.
   * 2. Change font properties to \`inherit\` in Safari.
   */
  /**
   * Remove the inner border and padding of focus outlines in Firefox.
   */
  /* Interactive
   */
  /*
   * Add the correct display in Edge and IE.
   */
  /*
   * Add the correct display in all browsers.
   */
  /* Scripting
   */
  /**
   * Add the correct display in IE 9-.
   */
  /**
   * Add the correct display in IE.
   */
  /* User interaction
   */
  /*
   * 1. Remove the tapping delay in IE 10.
   * 2. Remove the tapping delay on clickable elements
        in all browsers (opinionated).
   */
  /**
   * Add the correct display in IE 10-.
   */
  /* Accessibility
   */
  /**
   * Change the cursor on busy elements in all browsers (opinionated).
   */
  /*
   * Change the cursor on control elements in all browsers (opinionated).
   */
  /*
   * Change the cursor on disabled, not-editable, or otherwise
   * inoperable elements in all browsers (opinionated).
   */
  /*
   * Change the display on visually hidden accessible elements
   * in all browsers (opinionated).
   */
  font-family: sans-serif;
  font-family: var(--font-family-default);
  font-weight: var(--font-weight-default);
  font-size: 1rem;
  font-style: normal;
  line-height: 1.5rem;
  color: var(--color-black-80, #333);
  -webkit-tap-highlight-color: rgba(0, 0, 0, 0);
  font-variant-numeric: lining-nums;
  font-feature-settings: \\"lnum\\";
  font-size: 1em;
  line-height: 1.5em; }
  .dnb-form-status *,
  .dnb-form-status ::before,
  .dnb-form-status ::after {
    background-repeat: no-repeat;
    /* 1 */
    box-sizing: border-box;
    /* 2 */ }
  .dnb-form-status ::before,
  .dnb-form-status ::after {
    text-decoration: inherit;
    /* 1 */
    vertical-align: inherit;
    /* 2 */ }
  .dnb-form-status hr {
    height: 0;
    /* 1 */
    overflow: visible;
    /* 2 */ }
  .dnb-form-status main {
    display: block; }
  .dnb-form-status nav ol,
  .dnb-form-status nav ul {
    list-style: none; }
  .dnb-form-status pre {
    font-family: Menlo, Consolas, Roboto Mono, Ubuntu Monospace, Noto Mono, Oxygen Mono, Liberation Mono, monospace;
    /* 1 */
    font-size: 1em;
    /* 2 */ }
  .dnb-form-status a {
    background-color: transparent; }
  .dnb-form-status abbr[title] {
    text-decoration: underline;
    text-decoration: underline dotted; }
  .dnb-form-status b,
  .dnb-form-status strong {
    font-weight: bolder; }
  .dnb-form-status code,
  .dnb-form-status kbd,
  .dnb-form-status samp {
    font-family: Menlo, Consolas, Roboto Mono, Ubuntu Monospace, Noto Mono, Oxygen Mono, Liberation Mono, monospace;
    /* 1 */
    font-size: 1em;
    /* 2 */ }
  .dnb-form-status ::-moz-selection {
    background-color: #b3d4fc;
    /* 1 */
    color: #000;
    /* 1 */
    text-shadow: none; }
  .dnb-form-status audio,
  .dnb-form-status canvas,
  .dnb-form-status iframe,
  .dnb-form-status img,
  .dnb-form-status svg,
  .dnb-form-status video {
    vertical-align: middle; }
  .dnb-form-status audio,
  .dnb-form-status video {
    display: inline-block; }
  .dnb-form-status audio:not([controls]) {
    display: none;
    height: 0; }
  .dnb-form-status img {
    border-style: none; }
  .dnb-form-status svg:not([fill]) {
    fill: currentColor; }
  .dnb-form-status svg:not(:root) {
    overflow: hidden; }
  .dnb-form-status table {
    border-collapse: collapse; }
  .dnb-form-status button,
  .dnb-form-status input,
  .dnb-form-status select,
  .dnb-form-status textarea {
    font-family: inherit;
    font-size: inherit;
    line-height: inherit; }
  .dnb-form-status button,
  .dnb-form-status input,
  .dnb-form-status select {
    margin: 0; }
  .dnb-form-status button {
    overflow: visible;
    /* 1 */
    text-transform: none;
    /* 2 */ }
  .dnb-form-status button,
  .dnb-form-status [type='button'],
  .dnb-form-status [type='reset'],
  .dnb-form-status [type='submit'] {
    -webkit-appearance: button; }
  .dnb-form-status fieldset {
    padding: 0.35rem 0.75rem 0.625rem; }
  .dnb-form-status input {
    overflow: visible;
    word-break: normal;
    /* 1 */ }
  .dnb-form-status legend {
    color: inherit;
    /* 2 */
    display: table;
    /* 1 */
    max-width: 100%;
    /* 1 */
    white-space: normal;
    /* 1 */ }
  .dnb-form-status progress {
    display: inline-block;
    /* 1 */
    vertical-align: baseline;
    /* 2 */ }
  .dnb-form-status select {
    text-transform: none; }
  .dnb-form-status textarea {
    overflow: auto;
    /* 2 */
    resize: vertical;
    /* 3 */ }
  .dnb-form-status [type='checkbox'],
  .dnb-form-status [type='radio'] {
    padding: 0; }
  .dnb-form-status [type='search'] {
    -webkit-appearance: textfield;
    /* 1 */
    outline-offset: -2px;
    /* 2 */ }
  .dnb-form-status ::-webkit-inner-spin-button,
  .dnb-form-status ::-webkit-outer-spin-button {
    height: auto; }
  .dnb-form-status ::-webkit-input-placeholder {
    color: inherit;
    opacity: 0.54; }
  .dnb-form-status ::-webkit-search-decoration {
    -webkit-appearance: none; }
  .dnb-form-status ::-webkit-file-upload-button {
    -webkit-appearance: button;
    /* 1 */
    font: inherit;
    /* 2 */ }
  .dnb-form-status ::-moz-focus-inner {
    border-style: none;
    padding: 0; }
  .dnb-form-status details {
    display: block; }
  .dnb-form-status summary {
    display: list-item; }
  .dnb-form-status canvas {
    display: inline-block; }
  .dnb-form-status template {
    display: none; }
  .dnb-form-status a,
  .dnb-form-status area,
  .dnb-form-status button,
  .dnb-form-status input,
  .dnb-form-status label,
  .dnb-form-status select,
  .dnb-form-status summary,
  .dnb-form-status textarea,
  .dnb-form-status [tabindex] {
    -ms-touch-action: manipulation;
    /* 1 */
    touch-action: manipulation;
    /* 2 */ }
  .dnb-form-status [hidden] {
    display: none; }
  .dnb-form-status [aria-busy='true'] {
    cursor: progress; }
  .dnb-form-status [aria-controls] {
    cursor: pointer; }
  .dnb-form-status [aria-disabled='true'],
  .dnb-form-status [disabled] {
    cursor: not-allowed; }
  .dnb-form-status [aria-hidden='false'][hidden]:not(:focus) {
    clip: rect(0, 0, 0, 0);
    display: inherit;
    position: absolute; }
  @media print {
    .dnb-form-status {
      background-color: white; } }

/*
 * FormStatus component
 *
 */
.dnb-form-status {
  width: auto;
  display: flex;
  justify-content: flex-start;
  align-items: center;
  min-height: var(--input-height);
  border-radius: var(--input-border-radius);
  line-height: 0.5rem; }
  .dnb-form-status--text {
    padding: calc(0.5rem / 2) 1rem;
    cursor: text;
    color: var(--color-emerald-green);
    line-height: 1.5rem; }
  .dnb-icon + .dnb-form-status--text {
    padding-left: 0.5rem; }
  .dnb-form-status > .dnb-icon {
    display: flex;
    justify-content: center;
    align-items: center;
    width: 1.5em;
    height: 1.5em;
    margin-left: 0.5rem;
    color: inherit;
    font-size: 1rem;
    border-radius: 50%;
    border: 1px solid;
    border-color: currentColor; }
  .dnb-form-status[hidden] {
    display: none; }
  .dnb-form-status--fade-in {
    height: 0;
    opacity: 0;
    animation: fade-in 600ms ease-out 1 200ms forwards; }

@keyframes fade-in {
  from {
    opacity: 0;
    height: 0; }
  to {
    opacity: 1;
    height: var(--input-height); } }
"
`;
